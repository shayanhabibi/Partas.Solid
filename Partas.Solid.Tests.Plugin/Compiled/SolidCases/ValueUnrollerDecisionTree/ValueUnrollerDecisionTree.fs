module Partas.Solid.Tests.Plugin.Compiled.SolidCases.ValueUnrollerDecisionTree.ValueUnrollerDecisionTree

open Partas.Solid
open Fable.Core
open Fable.Core.JS
open Fable.Core.JsInterop

[<Erase>]
type Lib =
    [<ImportMember("tailwind-merge")>]
    static member twMerge(classes: string) : string = jsNative

    [<ImportMember("clsx")>]
    static member clsx(classes: obj) : string = jsNative

    [<CompiledName("cn")>]
    static member cn(classes: string array) : string =
        classes
        |> Lib.clsx
        |> Lib.twMerge

[<StringEnum>]
type Variant =
    | Black
    | Brown
    | Green

[<Erase>]
type ValueUnrollTest() =
    interface RegularNode

    [<DefaultValue>]
    val mutable variant: Variant

    [<SolidTypeComponent>]
    member props.__ =
        div (
            class' =
                Lib.cn
                    [| "bg-primary"
                       match props.variant with
                       | Brown -> "brown"
                       | Black -> "black"
                       | Green -> "green" |]
        )

[<StringEnum>]
type AltVariant =
    | Blue
    | Orange
    | Yellow

[<Erase>]
type ValueUnrollNestedTest() =
    interface RegularNode

    [<DefaultValue>]
    val mutable variant: Variant

    [<DefaultValue>]
    val mutable altVariant: AltVariant

    [<SolidTypeComponent>]
    member props.__ =
        div (
            class' =
                Lib.cn
                    [| "bg-primary"
                       match props.variant with
                       | Brown -> "brown"
                       | Black when props.altVariant = Blue -> "black & blue"
                       | Black -> "black"
                       | Green ->
                           match props.altVariant with
                           | Blue -> "green & blue"
                           | Orange -> "orange & blue"
                           | Yellow -> "yellow & blue" |]
        )
