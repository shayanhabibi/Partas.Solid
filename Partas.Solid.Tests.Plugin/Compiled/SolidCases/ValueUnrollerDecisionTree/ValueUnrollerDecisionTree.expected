
import { twMerge } from "tailwind-merge";
import { clsx } from "clsx";
import { splitProps } from "solid-js";

export function Lib_cn_Z35CD86D0(classes) {
    return twMerge(clsx(classes));
}

export function ValueUnrollTest(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <div class={Lib_cn_Z35CD86D0(["bg-primary", (PARTAS_LOCAL.variant === "black") ? "black" : ((PARTAS_LOCAL.variant === "green") ? "green" : "brown")])} />;
}

export function ValueUnrollNestedTest(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <div class={Lib_cn_Z35CD86D0(["bg-primary", (PARTAS_LOCAL.variant === "black") ? ((props.altVariant === "blue") ? "black & blue" : "black") : ((PARTAS_LOCAL.variant === "green") ? ((PARTAS_LOCAL.altVariant === "orange") ? "orange & blue" : ((PARTAS_LOCAL.altVariant === "yellow") ? "yellow & blue" : "green & blue")) : "brown")])} />;
}

